kin/monitors.py

```diff
@@ -1,227 +1,68 @@
-        """
-        Stop monitoring the account.
-
-        The thread will terminate in up to X seconds, where X is the timeout set by the blockchain.
-        """
-
-        # Set the stop event, this will terminate the thread once we get the next event from the blockchain.
-        self.stop_event.set()
-
-        # Change the retry value,
-        # this will cause an exception when trying to reconnect after timeout by the blockchain,
-        # which will terminate the thread.
-        self.sse_client.retry = None
-
-
-class MultiMonitor:
-    """Multi Monitor to monitor Kin payment on a multiple accounts"""
-
-    def __init__(self, kin_client, addresses, callback_fn):
-        """
-        Monitors multiple accounts for kin payments
-        :param kin.KinClient kin_client: a kin client directed to the correct network
-        :param str addresses: addresses to watch
-        :param callback_fn: the function to call on each received payment as `callback_fn(address, tx_data, monitor)`.
-        :type: callable[str,kin.transactions.SimplifiedTransaction ,kin.monitors.MultiMonitor]
-        """
-        self.kin_client = kin_client
-        self.callback_fn = callback_fn
-
-        if not addresses:
-            raise ValueError('no address to monitor')
-
-        for address in addresses:
-            if not is_valid_address(address):
-                raise StellarAddressInvalidError('invalid address: {}'.format(address))
-            if not self.kin_client.does_account_exists(address):
-                raise AccountNotFoundError(address)
-
-        self.addresses = addresses
-
-        # Currently, due to nonstandard SSE implementation in Horizon,
-        # using cursor=now will block until the first tx happens.
-        # Instead, we determine the cursor ourselves.
-        # Fix will be for horizon to send any message just to start a connection
-        params = {}
-        reply = self.kin_client.horizon.transactions(params={'order': 'desc', 'limit': 1})
-        if len(reply['_embedded']['records']) == 1:
-            cursor = reply['_embedded']['records'][0]['paging_token']
-            params = {'cursor': cursor}
-
-        # make synchronous SSE request (will raise errors in the current thread)
-        self.sse_client = self.kin_client.horizon.transactions(sse=True, params=params)
-
-        self.stop_event = Event()
-        # start monitoring thread
-        self.thread = Thread(target=self.event_processor, args=(self.stop_event,))
-        self.thread.daemon = True
-        self.thread.start()
-
-    def event_processor(self, stop_event):
-        """
-        Method to filter through SSE events and find kin payments for an account
-        :param threading.Event stop_event: an event that can be used to stop this method
-        """
-        import json
+            tx_data = SimplifiedTransaction(RawTransaction(tx))
+        except CantSimplifyError:
+            logger.debug("SSE transaction couldn't be simplified: ", tx)
```

Comment:
what's exactly this logger? is it controllable by the developer?

Reply:
Just a logger like any other. And yes, you can set it's level

See https://github.com/kinecosystem/migration-server/blob/master/scripts/src/init.py#L42